hal/watchdog_api.h

```diff
@@ -0,0 +1,156 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_API_H
+#define MBED_WATCHDOG_API_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef struct
+{
+  /**
+  * Refresh value for the watchdog in milliseconds. The maximum value of this
+  * setting is platform dependent, to find the maximum value for the current
+  * platform call hal_watchdog_get_max_timeout(void)
+  */
+  uint32_t timeout_ms;
+  /**
+   * Configures the watchdog for windowed mode of operation instead of running
+   * the independent watchdog. In this mode the a restriction is placed on the
+   * the time period in which the watchdog can be refreshed/kicked. If the
+   * watchdog is kicked too soon after it has last been refreshed the system
+   * will be reset. The period of time in which the reset will be triggered is
+   * defined by window_ms. This value is false by default.
+   */
+  bool enable_window;
+  /**
+   * Specifies the time window for the watchdog window in milliseconds. If the
+   * watchdog is configured to run in windowed mode kicking the watchdog less
+   * than this many milliseconds after it has last been kicked will trigger a
+   * system reset. This value must be less than timeout_ms.
+   */
+  uint32_t window_ms;
+  /**
+   * Configures the watchdog timer to run while the core is in sleep mode. By
+   * default when the system is put into the sleep the watchdog timer is paused.
+   * Enabling this setting causes the timer to countdown during this time. This
+   * flag is disabled by default.
+   */
+  bool enable_sleep;
+} watchdog_config_t;
+
+
+typedef enum {
+  WATCHDOG_STATUS_OK,
+  WATCHDOG_STATUS_NOT_SUPPORTED,
+  WATCHDOG_STATUS_INVALID_ARGUMENT
+} watchdog_status_t;
+
+#ifdef __cplusplus
+    extern "C" {
+#endif
+
+/** Initialise and start a watchdog timer with the given configuration.
+ *
+ * The watchdog timer initialises a timer with a period specified by
```

Comment:
That's great stuff, but could you move the general operation description to the module/file doxy rather than init.